File: cpp/merge_sorted_array.cpp
class Solution {
public:
    void merge(int A[], int m, int B[], int n) {
        int ia = m - 1;
        int ib = n - 1;
        int index = m + n -1;
        while (ia >= 0 && ib >= 0)
        {
            if (A[ia] < B[ib])
                A[index--] = B[ib--];
            else
                A[index--] = A[ia--];
        }
        while (ib >= 0)
            A[index--] = B[ib--];
    }
};

File: python/merge_sorted_array.py
class Solution:
    # @param A  a list of integers
    # @param m  an integer, length of A
    # @param B  a list of integers
    # @param n  an integer, length of B
    # @return nothing
    def merge(self, A, m, B, n):
        ia = m - 1
        ib = n - 1
        index = m + n - 1
        while ia >= 0 and ib >= 0:
            if A[ia] < B[ib] or ia ==-1:
                A[index] = B[ib]
                index -= 1
                ib -= 1
            else:
                A[index] = A[ia]
                index -= 1
                ia -= 1
        while ib >= 0:
            A[index] = B[ib]
            index -= 1
            ib -= 1
        return A